### DISCUSSION TITLE
Usage for registerNewSymbolNameProvider

### DISCUSSION THREAD

**Discussion by zeyugao** [microsoft/vscode-discussions#1607] [Extension Development QnA] [answered]
I am trying to developing a extension to suggest the symbol renaming. Just like the screen shot

<img width="355" alt="BF8168AB-CB6C-4EB0-83A2-C048A5DF9820" src="https://github.com/user-attachments/assets/55dc6a28-1a93-4453-bcac-5d21cd524800">

After digging the source code how vscode handle the rename suggestion, I found that it depends on newSymbolNamesProvider (https://github.com/microsoft/vscode/commit/89d33f932bd862024b7d168870f33b5bf3804583#diff-344d9a04981d1b752b3151fcf03926d317dc6971660120fe3abf1078187746dcR214). It seems that it can be registered with `registerNewSymbolNameProvider`, and it is also mentioned in the release notes (https://code.visualstudio.com/updates/v1_87#_new-symbol-names-provider-api).

But after glancing around usage of that api (through: https://github.com/search?q=registerNewSymbolNameProvider&type=code), it seems that there is little extension using it. And it is also not exists in the npm package `@types/vscode`. So, I wonder how I can use this api to suggest the symbol by my own vscode extension (It seems that the github copilot has done it).

**Comment by isidorn** [comment, marked as answer]
Currently this is not exposed through the finalised API. I will let you know when there are updates here.
